python/examples/test_pyreduction.py

```diff
@@ -0,0 +1,65 @@
+import sys, os
+import math
+from vowpalwabbit import pyvw
+
+class NoopPythonReduction(pyvw.Copperhead):
+    def _predict(self):
+        print("hello there I'm predicting stuff")
+
+    def _learn(self, ec, learner):
+        # print("hello there I can also learn stuff btw the total_sum_feat_sq is " + str(ec.get_total_sum_feat_sq()))
+        learner.learn(ec)
+
+# this is a recreation of the impl in vw
+# see https://github.com/VowpalWabbit/vowpal_wabbit/blob/ac3a2c21a9760b68ce49368b11a35bf95faeb8b8/vowpalwabbit/binary.cc
+class BinaryPythonReduction(pyvw.Copperhead):
+    def _predict(self):
+        print("hello there I'm predicting stuff")
+
+    def _learn(self, ec, learner):
+        learner.learn(ec)
+
+        if ec.get_simplelabel_prediction() > 0:
+            ec.set_simplelabel_prediction(1)
+        else:
+            ec.set_simplelabel_prediction(-1)
+
+        temp_label = ec.get_simplelabel_label()
+
+        # going to have to expose FLT_MAX?
+        if temp_label != sys.float_info.max:
+            if math.fabs(temp_label) != 1.0:
+                print("You are using label " + temp_label << " not -1 or 1 as loss function expects!")
```

Comment:
if using python3.6 you can format strings using `f` e.g.: `print(f"You are using label {temp_label}")`